### PR TITLE
Fixed numerous issues with sharing state across the network.

### LINKED ISSUE #164: Playbar Song Incorrect (Guest)
The "now playing" song label is incorrect after rotation on the guest phone

### LINKED ISSUE #169: Now Playing Icon (Guest)
Now Playing Icon does not show up on the guest phone

### LINKED ISSUE #176: Current song icon not displayed on guest device
Observed on Nexus 7, on Test 12b.

### LINKED ISSUE #180: Playbar not correct immediately after joining an existing network.
After performing Test 15, the playbar on the guest showed "Now Playing".  The correct state was displayed after the song was paused on the host.

### PR DESCRIPTION
**Pull request by theJenix**
Modified connection workflow so the host transmits a play status message to
connecting guests.  Guests then set the currentEntry member of PlaylistService
and dispatch a play or paused intent to notify the rest of the system.

Refactored PlaybarFragment to listen only to the intents from playlist service
(not the messages), and pull current song data from the playlist.

Refactored RemoteAudioPlayer to listen to intents from the playlist service
(instead of messages) and update the internal state accordingly.

Fixed a bug where the focus changed listener was being left registered when we
recreated an AudioPlayerWithEvents object, leading to weird play/pause behavior.

NOTE: This also seems to have addressed the issue of showing the "now playing" icon on the guest.

fixes #164 
fixes #180 
fixes #176 
fixes #169

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/service/PlaylistService.java [hunk 1]
```diff
@@ -227,6 +233,31 @@ public void onReceiveAction(Context context, Intent intent) {
...
+        })
+        .addLocalAction(MessagingService.ACTION_PLAY_STATUS_MESSAGE, new IBroadcastActionHandler() {
+
+            @Override
+            public void onReceiveAction(Context context, Intent intent) {
+                boolean isPlaying = intent.getBooleanExtra(MessagingService.EXTRA_IS_PLAYING, false);
+                String macAddress = intent.getStringExtra(MessagingService.EXTRA_ADDRESS);
+                long   songId     = intent.getLongExtra(  MessagingService.EXTRA_SONG_ID,
+                                                          SongMetadata.UNKNOWN_SONG);
+                int   entryId     = intent.getIntExtra(  MessagingService.EXTRA_ENTRY_ID, 0);
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/service/PlaylistService.java:249]: If you are going to line these up line them up

**Comment by reidbaker** [comment]
One formatting nit. @theJenix fix it in some other branch I am pulling this in

**Comment by reidbaker** [comment]
HEY @TheLastCrusade/owners check this out I was a little over zelous and pulled it in after being the only one to look at it.

**Comment by dgreenhalgh** [comment]
@reidbaker Any specific concerns or are you just thinking you looked through it too quickly?

**Pull request merged**